src/node_file.cc

```diff
@@ -138,9 +123,28 @@ void FSReqWrap::Resolve(Local<Value> value) {
   MakeCallback(env()->oncomplete_string(), arraysize(argv), argv);
 }
 
+void FSReqWrap::Init(const char* syscall,
+                     const char* data,
+                     enum encoding encoding,
+                     Ownership ownership) {
+  CHECK_EQ(data_, nullptr);  // Initialize only once
+  syscall_ = syscall;
+  encoding_ = encoding;
+
+  const bool copy = (data != nullptr && ownership == Ownership::COPY);
+  if (copy) {
+    size_t size = 1 + strlen(data);
```

Comment:
Same issue as `strdup(data)`, not binary data-safe.  With the recent changes to this file and the many layers of templates I have a hard time telling whether that's actually safe now.

It didn't use to be (`data` could be a pointer to the buffer argument to `fs.write()`) but now it might be.  In that case it should be renamed to e.g. `dest`, for parity with `AsyncCall`.

If `data` is actually `dest` then there is not much point in the whole ownership management thing because `dest` is a few bytes to a few kB worst case.

Reply:
That's kind of what I'm thinking too really (re: the ownership management piece). I may tweak this a bit more to get rid of `Ownership` entirely and just copy every time and see what the perf impact will be.

Reply:
@jasnell If I understand your comment correctly and you’re thinking that we’re allocating only a few bytes in most cases anyway, how about using a `MaybeStackBuffer` for this? The name isn’t totally accurate anymore but it’s kind of what it’s there for.

Reply:
Yep. That's exactly where I was going.